_data-prepper/pipelines/configuration/sinks/s3.md

```diff
@@ -119,72 +150,50 @@ Option | Required | Type | Description
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
-`path_prefix` | No | String | The S3 key prefix path to use for objects written to S3. Accepts date-time formatting. For example, you can use `%{yyyy}/%{MM}/%{dd}/%{HH}/` to create hourly folders in S3. The prefix path should end with `/`. By default, Data Prepper writes objects to the root of the S3 bucket.
-
+`path_prefix` | No | String | The S3 key prefix path to use for objects written to S3. Accepts date-time formatting and dynamic injection of values using [Data Prepper expressions](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/). For example, you can use `/${/my_partition_key}/%{yyyy}/%{MM}/%{dd}/%{HH}/` to create hourly folders in S3 based on the `my_partition_key` value. The prefix path should end with `/`. By default, Data Prepper writes objects to the S3 bucket root.
 
-## codec
+## `codec`
 
 The `codec` determines how the `s3` source formats data written to each S3 object.
 
-### avro codec
+### `avro` codec
 
 The `avro` codec writes an event as an [Apache Avro](https://avro.apache.org/) document.
 
-Because Avro requires a schema, you may either define the schema yourself, or Data Prepper will automatically generate a schema.
-In general, you should define your own schema because it will most accurately reflect your needs.
-
-We recommend that you make your Avro fields use a null [union](https://avro.apache.org/docs/current/specification/#unions).
-Without the null union, each field must be present or the data will fail to write to the sink.
-If you can be certain that each each event has a given field, you can make it non-nullable.
+Because Avro requires a schema, you may either define the schema yourself or have Data Prepper automatically generate it. It is recommended that you define your schema to accurately reflect your needs.
 
-When you provide your own Avro schema, that schema defines the final structure of your data.
-Therefore, any extra values inside any incoming events that are not mapped in the Arvo schema will not be included in the final destination.
-To avoid confusion between a custom Arvo schema and the `include_keys` or `exclude_keys` sink configurations, Data Prepper does not allow the use of the `include_keys` or `exclude_keys` with a custom schema.
+We recommend that you make your Avro fields use a null [union](https://avro.apache.org/docs/current/specification/#unions). Without the null union, each field must be present or the data will fail to write to the sink. If you can be certain that each each event has a given field, you can make it non-nullable.
```

Comment:
```suggestion
It is recommended making Avro fields use a null [union](https://avro.apache.org/docs/current/specification/#unions) to allow missing values; otherwise, ensure all required fields are present for each event. Use non-nullable fields only when certain of their presence.
```